content/docs/user-guide/data-pipelines/defining-pipelines.md

```diff
@@ -0,0 +1,253 @@
+# Defining Pipelines
+
+Pipelines represent data workflows that you want to **reproduce** reliably -- so
+the results are consistent. The regular pipelining process involves:
+
+- Obtain and `dvc add` the project's initial data requirements (see [Data
+  Management]). This <abbr>caches</abbr> the data and generates `.dvc` files.
+
+- Define the pipeline [stages](#stages) in `dvc.yaml` files (more details
+  later). Example structure:
+
+  ```yaml
+  stages:
+    prepare: ... # stage 1 definition
+    train: ... # stage 2 definition
+    evaluate: ... # stage 3 definition
+  ```
+
+- Capture other useful metadata such as runtime
+  [parameters](#parameter-dependencies), performance [metrics], and [plots] to
+  visualize. DVC supports multiple file formats for these.
+
+<admon type="info">
+
+We call this file-based definition _codification_ (YAML format in our case). It
+has the added benefit of allowing you to develop pipelines on standard Git
+workflows ([GitOps]).
+
+[gitops]: /doc/use-cases/versioning-data-and-model-files
+
+</admon>
+
+Stages usually take some data and run some code, producing an output (e.g. an ML
+model). The pipeline is formed by making them interdependent, meaning that the
+output of a stage becomes the input of another, and so on. Technically, this is
+called a _dependency graph_ (DAG).
+
+[data management]: /doc/start/data-management
+[metrics]: /doc/command-reference/metrics
+[plots]: /doc/user-guide/visualizing-plots
+
+## Directed Acyclic Graph (DAG)
+
+DVC represents a pipeline internally as a _graph_ where the nodes are stages and
+the edges are _directed_ dependencies (e.g. A before B). And in order for DVC to
+run a pipeline, its topology should be _acyclic_ -- because executing cycles
+(e.g. A -> B -> C -> A ...) would continue indefinitely. [More about DAGs].
+
+<admon type="info">
+
+Stage execution will be determined entirely by the DAG, not by the order in
+which stages are found in `dvc.yaml`.
+
+</admon>
+
+Note that while each pipeline is a DAG, this doesn't mean a single `dvc.yaml`
```

Comment:
this is out of place, should be a bit more general somewhere

Reply:
Hm... OK I moved it up but it feels like the intro has too much text now. I guess if it's "book format" then it's fine.